yarn-project/prover-client/src/orchestrator/proving-state.ts

```diff
@@ -0,0 +1,194 @@
+import { ProcessedTx, ProvingResult } from '@aztec/circuit-types';
+import {
+  BaseOrMergeRollupPublicInputs,
+  Fr,
+  GlobalVariables,
+  NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP,
+  Proof,
+  RootParityInput,
+} from '@aztec/circuits.js';
+import { randomBytes } from '@aztec/foundation/crypto';
+import { Tuple } from '@aztec/foundation/serialize';
+
+/**
+ * Enums and structs to communicate the type of work required in each request.
+ */
+export enum PROVING_JOB_TYPE {
+  STATE_UPDATE,
+  BASE_ROLLUP,
+  MERGE_ROLLUP,
+  ROOT_ROLLUP,
+  BASE_PARITY,
+  ROOT_PARITY,
+}
+
+export type ProvingJob = {
+  type: PROVING_JOB_TYPE;
+  operation: () => Promise<void>;
+};
+
+export type MergeRollupInputData = {
+  inputs: [BaseOrMergeRollupPublicInputs | undefined, BaseOrMergeRollupPublicInputs | undefined];
+  proofs: [Proof | undefined, Proof | undefined];
+};
+
+/**
+ * The current state of the proving schedule. Contains the raw inputs (txs) and intermediate state to generate every constituent proof in the tree.
+ * Carries an identifier so we can identify if the proving state is discarded and a new one started.
+ * Captures resolve and reject callbacks to provide a promise base interface to the consumer of our proving.
+ */
+export class ProvingState {
+  private _stateIdentifier: string;
```

Comment:
nit: I'm not keen on the underscore prefix. I don't know what it "means" in our codebase.

Reply:
+1!